network/linux_s/linux_s.c

```diff
@@ -0,0 +1,300 @@
+	.socket_disconnect = (int32_t (*)(void *, uint32_t))linux_socket_disconnect,
+	.socket_send = (int32_t (*)(void *, uint32_t, const void *, uint32_t))linux_socket_send,
+	.socket_recv = (int32_t (*)(void *, uint32_t, void *, uint32_t))linux_socket_recv,
+	.socket_sendto = (int32_t (*)(void *, uint32_t, const void *, uint32_t, const struct socket_address* to))linux_socket_sendto,
+	.socket_recvfrom = (int32_t (*)(void *, uint32_t, void *, uint32_t, struct socket_address* from))linux_socket_recvfrom,
+	.socket_bind = (int32_t (*)(void *, uint32_t, uint16_t))linux_socket_bind,
+	.socket_listen = (int32_t (*)(void *, uint32_t, uint32_t))linux_socket_listen,
+	.socket_accept= (int32_t (*)(void *, uint32_t, uint32_t*))linux_socket_accept
+};
+
+/**
+ * @brief Get network interface reference
+ * @param desc - Socket descriptor
+ * @param net - Address where to store the reference to the network interface
+ * @return
+ *  - \ref SUCCESS : On success
+ */
+int32_t linux_get_network_interface(struct linux_desc *desc,
+				    struct network_interface** net)
+{
+	*net = &linux_net;
+
+	return SUCCESS;
+}
+
+/** @brief See \ref network_interface.socket_open */
+static int32_t linux_socket_open(struct linux_desc *desc, uint32_t *sock_id,
+				 enum socket_protocol prot, uint32_t buff_size)
+{
+	*sock_id = socket(AF_INET,SOCK_STREAM,IPPROTO_TCP);
+	int flags = fcntl(*sock_id, F_GETFL);
+	fcntl(*sock_id, F_SETFL, flags | O_NONBLOCK);
+
+	if(*sock_id < 0)
+		return *sock_id;
+	return SUCCESS;
+}
+
+/** @brief See \ref network_interface.socket_close */
+static int32_t linux_socket_close(struct linux_desc *desc, uint32_t sock_id)
+{
+	int32_t ret;
+	ret = close(sock_id);
+	if(ret < 0)
+		return -errno;
+	return ret;
+}
+
+/** @brief See \ref network_interface.socket_connect */
+static int32_t linux_socket_connect(struct linux_desc *desc, uint32_t sock_id,
+				    struct socket_address *addr)
+{
+	int32_t ret;
+	struct sockaddr_in saddr;
+	memset(&saddr,0,sizeof(saddr));
+	saddr.sin_family = AF_INET;
+	saddr.sin_port = htons(addr->port);
+	struct hostent* hptr = gethostbyname(addr->addr);
+	saddr.sin_addr.s_addr = ((struct in_addr*) hptr->h_addr_list[0])->s_addr;
+
+	socklen_t len = sizeof(saddr);
+
+	int flags = fcntl(sock_id, F_GETFL);
+	fcntl(sock_id, F_SETFL, flags | O_NONBLOCK);
+	ret = connect(sock_id,(struct sockaddr*) &saddr,len);
+	if(ret < 0)
+		return -errno;
+	return ret;
+
+}
+
+/** @brief See \ref network_interface.socket_disconnect */
+static int32_t linux_socket_disconnect(struct linux_desc *desc,
+				       uint32_t sock_id)
+{
+	int32_t ret;
+
+	ret = linux_socket_close(desc, sock_id);
+	if(ret < 0)
+		return -errno;
+	return SUCCESS;
+}
+
+/** @brief See \ref network_interface.socket_send */
+static int32_t linux_socket_send(struct linux_desc *desc, uint32_t sock_id,
+				 const void *data, uint32_t size)
+{
+	int32_t ret;
+
+	int flags = fcntl(sock_id, F_GETFL);
+	fcntl(sock_id, F_SETFL, flags | O_NONBLOCK);
```

Comment:
Do a call to fcntl is needs to be done in each send and recv ? 
Isn't enough in open and accept? Because is the same sock_id.